src/geofabrics/lidar_fetch.py

```diff
@@ -0,0 +1,155 @@
+                 redownload_files: bool = False, download_limit_gbytes: typing.Union[int, float] = 100, verbose: bool = False):
+        """ Load in lidar with relevant processing chain.
+        
+        Note in case of multiple datasets could select by name, 
+        spatial extent, or most recent. download_size is in GB. """
+        
+        self.catchment_geometry = catchment_geometry
+        self.cache_path = pathlib.Path(cache_path)
+        self.redownload_files = redownload_files
+        self.download_limit_gbytes = download_limit_gbytes
+        self.verbose = verbose
+        
+        self._dataset_prefixes = None
+        
+    def run(self):
+        """ Querey for  LiDAR data within a catchment and download any that hasn't already been downloaded """
+        self._dataset_prefixes = []
+        
+        json_response = self.query_for_datasets_inside_catchment()
+        
+        ot_endpoint_url = urllib.parse.urlunparse((self.SCHEME, self.NETLOC_DATA, "", "", "", ""))
+        client = boto3.client('s3', endpoint_url=ot_endpoint_url, 
+                              config=botocore.config.Config(signature_version=botocore.UNSIGNED))
+        
+        # cycle through each dataset within a region
+        for i in range(len(json_response['Datasets'])):
+            dataset_prefix = json_response['Datasets'][i]['Dataset']['alternateName']
+            self._dataset_prefixes.append(dataset_prefix)
+            
+            tile_info = self._get_dataset_tile_info(client, dataset_prefix)
+            
+            # check download size limit is not exceeded
+            lidar_size_bytes = self._calculate_dataset_download_size(client, dataset_prefix, tile_info)
+            
+            assert lidar_size_bytes/1000/1000/1000 < self.download_limit_gbytes, "The size of the LiDAR to be " \
+                + "downloaded is greater than the specified download limit of " + str(self.download_limit_gbytes)
+            
+            # check for tiles and download as needed
+            self._download_tiles_in_catchment(client, dataset_prefix, tile_info)
+            
+        
+    def query_for_datasets_inside_catchment(self):
+        """ Function to check for data in search region using hte otCatalogue API
+        https://portal.opentopography.org/apidocs/#/Public/getOtCatalog """
+        
+        api_queary = {
+            "productFormat": "PointCloud",
+            "minx": self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds['minx'].min(),
+            "miny": self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds['miny'].min(),
+            "maxx": self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds['maxx'].max(),
+            "maxy": self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds['maxy'].max(),
+            "detail": False,
+            "outputFormat": "json",
+            "inlcude_federated": True
+        }
```

Comment:
```suggestion
        catchment_bounds = self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds
        api_query = {
            "productFormat": "PointCloud",
            "minx": catchment_bounds['minx'].min(),
            "miny": catchment_bounds['miny'].min(),
            "maxx": catchment_bounds['maxx'].max(),
            "maxy": catchment_bounds['maxy'].max(),
            "detail": False,
            "outputFormat": "json",
            "include_federated": True
        }
```
Fixing spelling errors, and reducing code complexity and reducing repearting of conversions to CRS

Might be worth adding a linter and spell checker that can automatically clean up files. I know some in PyCharm but we can discuss this next time we meet

Reply:
Great! I'd already updated the self.CRS to self.OT_CRS, hopefully improving readability, so patching your suggestion in on-top.